Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -162,3 +162,109 @@ instance : HasForget₂ ProfiniteGrp Grp where
   }
 
 end ProfiniteGrp
+
+/-!
+# The projective limit of finite groups is profinite
+
+* `ProfiniteGrp.limit` : the concretely constructed projective limit of finite groups
+  as a subgroup of the pi-type
+
+* `ofLimit`: projective limit of finite groups is a profinite group
+
+* Verify that the constructed limit satisfies the universal property.
+-/
+
+section Profiniteoflimit
+
+/- In this section, we prove that the projective limit of finite groups is profinite-/
+
+universe w w'
+
+namespace ProfiniteGrp
+
+variable {J : Type v} [SmallCategory J] (F : J ⥤ ProfiniteGrp.{max v w'})
+
+/-- Concretely constructing the limit of topological group as a subgroup of the  pi-type. -/
+def limit : Subgroup (Π j : J, F.obj j) where
+  carrier := {x | ∀ ⦃i j : J⦄ (π : i ⟶ j), F.map π (x i) = x j}
+  mul_mem' hx hy _ _ π := by simp only [Pi.mul_apply, map_mul, hx π, hy π]
+  one_mem' := by simp only [Set.mem_setOf_eq, Pi.one_apply, map_one, implies_true]
+  inv_mem' h _ _ π := by simp only [Pi.inv_apply, map_inv, h π]
+
+/--The functor mapping a profinite group to its underlying profinite space-/
+def profiniteGrpToProfinite : ProfiniteGrp ⥤ Profinite where
+  obj G := G.toProfinite
+  map f := ⟨f, by continuity⟩
+
+instance instProfiniteGrpToProfiniteFaithful : profiniteGrpToProfinite.Faithful := {
+  map_injective := fun {_ _} _ _ h =>
+    ConcreteCategory.hom_ext_iff.mpr (congrFun (congrArg ContinuousMap.toFun h)) }
+
+instance : CompactSpace (limit F) := inferInstanceAs
+  (CompactSpace (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v w'})).pt)
+
+/-- Making the direct limit of `FiniteGrp` into a `ProfiniteGrp`. -/
+def ofLimit : ProfiniteGrp := .of (ProfiniteGrp.limit F)
+
+/-- Verify that the limit constructed above exist projections to the `FiniteGrps`
+that are compatible with the morphisms between them. -/
+def limitCone : Limits.Cone F where
+  pt := ofLimit F
+  π :=
```

Comment:
```suggestion
  pt := ofProfinite (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v w'})).pt
  π :=
```

Reply:
should `ofLimit ` still exist as an abbrev? This would save some writing.

Reply:
I think `ofLimit` is unnecessary. Now you can access it via `(ProfiniteGrp.limitCone _).pt`

Reply:
But that would appear at a lot of places in the incoming PRs, I think this at least reduce writing and suffix (I don't think this would be easy to read)